src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -482,4 +488,77 @@ export class StudyViewPageStore {
         },
         default: {}
     });
+
+    readonly mutationCounts = remoteData<MutationCount[]>({
+        await:()=>[
+            this.samples,
+            this.mutationProfiles
+        ],
+        invoke:async()=>{
+            const studyToSamples = _.groupBy(this.samples.result!, s=>s.studyId);
+            return _.flatten(await Promise.all(
+                this.mutationProfiles.result!.map(mutationProfile=>{
+                    const samples = studyToSamples[mutationProfile.studyId];
+                    if (samples && samples.length) {
+                        return defaultClient.fetchMutationCountsInMolecularProfileUsingPOST({
+                            molecularProfileId: mutationProfile.molecularProfileId,
+                            sampleIds: samples.map(s=>s.sampleId)
+                        });
+                    } else {
+                        return Promise.resolve([]);
+                    }
+                })
+            ));
+        }
+    });
+
+    readonly fractionGenomeAltered = remoteData<FractionGenomeAltered[]>({
+        await:()=>[
+            this.samples
+        ],
+        invoke:async()=>{
+            const studyToSamples = _.groupBy(this.samples.result!, s=>s.studyId);
+            return _.flatten(await Promise.all(
+                _.map(studyToSamples, (samples, studyId)=>{
+                    if (samples && samples.length) {
```

Comment:
I think this condition is not necessary as `studyToSamples` comes from `groupBy`

Reply:
thats true but id rather not couple those